### PR TITLE
Android bugfix merge

### PR DESCRIPTION
**Pull request by mike-mh**
Branch includes several significant changes and bugfixes:
- DCAPP now broadcasts battery life remaining
- Fixed delay bug (changing the speed of broadcast caused delay)
- Fixed disconnection bug (device now attempts to reconnect forever when disconnected)
- Includes changes to schema and validation to include battery and publish rate

### CODE REVIEW

File: dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java [hunk 1]
```diff
@@ -270,11 +314,8 @@ public int onStartCommand(Intent intent, int flags, int startId) {
      *         should be disabled and the accelerationListener unregistered.
      */
     public void onDestroy() {
-        if (client != null) {
-            client.unregisterResources();
```
Comment (krxsky) [inline dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java:274]: Just curious, what does this call do?

**Comment by krxsky** [comment]
Had a small comment on the deletion of a line. Otherwise looks good to me.

**Comment by mike-mh** [comment]
@krxsky

Glad you brought that up. The method 'unregisterResources' kills the threads attributed to the MQTT client so that its operations won't persist in the background. David and I were having a ****\* of a time resolving it. The log was mentioning that there were service leaks (i.e. operations that continued to run in the background without stopping). Usually we got away with it because we weren't changing the settings enough. 

But... if you changed the settings enough you'd keep using more and more RAM for operations until the program crashes. It's almost like a memory leak... but for Java.

**Comment by mike-mh** [comment]
@CarlosBalladares 

We need to confirm that:
- When the user changes the IP address, watch disconnects if it's inappropriate
- When the user can reconnect after disconnecting (also make sure that the publish rate matches the setting on the watch)
- When the network is down, watch should disconnect and when it comes back online should reconnect automatically (to test this, turn the mosquitto broker off then on again)
- When the user changes the speed setting the publish rate changes instantaneously to the right value (no 'hickups' e.g. changing it to 400 ms but the value reads 800 ms at first)
- Ensure that batterry and publish_rate values are included in the Android Data payload

**Comment by CarlosBalladares** [comment]
I found a minor issue:
-Whenever the android client is disconnected by changing the ip address and reconnected to the server the frequency is set to 200ms , this value is displayed in the watch but the slider doesn't change .

screenshot_2010-01-01-07-25-49
- At some point after a some disconnects and reconnects, the app stopped connecting to the server. even when given a good ip. I'm not sure if this was a problem with my network or  with the app. I'll test more on this.

**Comment by mike-mh** [comment]
@CarlosBalladares 

Thanks for that Carlos. The '200 ms' issue should be resolved and it turns out there's a race condition when the user changes the IP address of the connection. In other words, the client works fine so long as you what about 2-3 seconds before changing the IP address, otherwise there's a significant delay due to the need to free up resources before a new connection can be made. I'm going to introduce a delay before the connection is made when the user changes the IP address and it should resolve the issue will keep you posted.

**Pull request merged**